pandas/core/strings.py

```diff
@@ -323,15 +328,24 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
+    case : boolean, default None
```

Comment:
I know using such a sentinel is a way to go when you want to be able to distinguish None (a passed None as a value for the arg you want to catch), but this is not the case here. We can use None to distinguish a passed True or the default None as True. We don't need to be able to catch an explicitly passed None. So I don't understand why using `object` is needed here? 
Further it has the disadvantage that you get an ugly value in the signature.

As far as I see it, we just need to choose between those options:

- keep default as True, and say that the argument is ignored if compiled regex is passed
- default of None, which is set to True if no compiled regex is passed, and we can raise if something else as None is passed in case of compiled regex.

Code-wise the first is the simplest. 
The second we can give more information to the user.